dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerOwnershipManager.java

```diff
@@ -237,7 +237,14 @@ private void validateGrantTemporaryOwnershipInputs(
   @Override
   @Transactional(readOnly = true)
   public boolean hasAccess(UserDetails user, TrackedEntity trackedEntity, Program program) {
-    if (canSkipOwnershipCheck(user, program) || trackedEntity == null) {
+    return hasAccess(user, trackedEntity, program, false);
+  }
+
+  @Override
+  @Transactional(readOnly = true)
+  public boolean hasAccess(
+      UserDetails user, TrackedEntity trackedEntity, Program program, boolean skipOwnershipCheck) {
```

Comment:
I would still remove this `skipOwnershipCheck` and have the caller decide if it needs to call this method or not based on the flag.